apps/runtime/internal/cacheintercept/cacheintercept_test.go

```
@@ -0,0 +1,211 @@
+// New must not keep the caller's URL: a later mutation of it would move the
+// forward target past the scheme and host validation.
+func TestUpstreamMutationAfterNewIsIgnored(t *testing.T) {
+	w := setup(t, http.StatusOK, "ok")
+	w.upstreamURL.Scheme = "http"
+	w.upstreamURL.Host = "wrong.invalid:1"
+
+	resp, err := w.client.Get("https://" + cacheca.CacheHost + "/")
+	if err != nil {
+		t.Fatalf("request failed after mutating the caller's URL: %v", err)
+	}
+	resp.Body.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `resp.Body.Close` is not checked

(errcheck)